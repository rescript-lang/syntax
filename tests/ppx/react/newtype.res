@react.component
let make = (type a, ~a: a, ~b, _) => <div />
