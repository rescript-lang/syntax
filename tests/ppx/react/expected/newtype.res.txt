@@jsxConfig({version: 3})

module V3 = {
  @obj
  external makeProps: (
    ~a: '\"type-a",
    ~b: array<option<[#Foo('\"type-a")]>>,
    ~c: 'a,
    ~key: string=?,
    unit,
  ) => {"a": '\"type-a", "b": array<option<[#Foo('\"type-a")]>>, "c": 'a} = ""

  @react.component
  let make = (type a, ~a: a, ~b: array<option<[#Foo(a)]>>, ~c: 'a, _) =>
    ReactDOMRe.createDOMElementVariadic("div", [])
  let make = {
    let \"Newtype$V3" = (
      \"Props": {"a": '\"type-a", "b": array<option<[#Foo('\"type-a")]>>, "c": 'a},
    ) => make(~c=\"Props"["c"], ~b=\"Props"["b"], ~a=\"Props"["a"])
    \"Newtype$V3"
  }
}

@@jsxConfig({version: 4, mode: "classic"})

module V4C = {
  type props<'a, 'b, 'c> = {key?: string, a: 'a, b: 'b, c: 'c}

  @react.component
  let make = ({a, b, c, _}: props<'\"type-a", array<option<[#Foo('\"type-a")]>>, 'a>) =>
    ReactDOMRe.createDOMElementVariadic("div", [])
  let make = {
    let \"Newtype$V4C" = (props: props<_>) => make(props)
    \"Newtype$V4C"
  }
}

@@jsxConfig({version: 4, mode: "automatic"})

module V4A = {
  type props<'a, 'b, 'c> = {key?: string, a: 'a, b: 'b, c: 'c}

  @react.component
  let make = ({a, b, c, _}: props<'\"type-a", array<option<[#Foo('\"type-a")]>>, 'a>) =>
    ReactDOM.jsx("div", {key: ?None})
  let make = {
    let \"Newtype$V4A" = (props: props<_>) => make(props)
    \"Newtype$V4A"
  }
}
